test/basiccontrollermap.c

```diff
@@ -0,0 +1,120 @@
+/*
+  Copyright (C) 1997-2022 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely.
+*/
+
+/* Game controller mapping generator */
+/* Gabriel Jacobo <gabomdq@gmail.com> */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "SDL.h"
+#include "testutils.h"
+
+#ifndef SDL_JOYSTICK_DISABLED
+
+
+int main(int argc, char *argv[]) {
+    SDL_Event event;
+    SDL_Joystick *joy;
+    const char *type;
+    char guid[64];
+
+    SDL_Init(SDL_INIT_JOYSTICK);
+    joy = SDL_JoystickOpen(0);
```

Comment:
I don't want to be "a joy breaker", but `joy` can be `NULL` :rofl:  (I'll show myself out)

The other examples test for `NULL` and log it appropriately.

Reply:
Thanks! Done!